src/main/java/com/fasterxml/jackson/databind/jsontype/TypeResolverBuilder.java

```diff
@@ -191,4 +191,11 @@ public default T withDefaultImpl(Class<?> defaultImpl) {
         //    possibly unsafe variant
         return defaultImpl(defaultImpl);
     }
+
+    /**
+     * Method for overriding type information.
+     * 
+     * @since 2.16
+     */
+    public T withSettings(JsonTypeInfo.Value typeInfo);
```

Comment:
Forgot to think it through -- this is backwards-incompatible change and breaks at least one format backend (Ion format). I'll add default implementation that produces runtime exception, but in hindsight my suggestion for backporting was bad one.

Reply:
Also should have noticed and pointed out that semantics are wrong wrt `withXxx()` methods which (in context of Jackson) should always either:

1. Return `this` as-is, with no modifications (usually operate on immutable types)
2. Construct a new object with different construction

and anything else should not use withXxx() naming.

This is my bad, not your fault.